Doc: Update documentation on usage with Webpack (#12823)
facebook/jest: docs/Webpack.md

---
id: webpack

> title: Using with Webpack

Reviewer: If it is spell with capitalized first letter here, the blog post section needs to be updated as well then. There are some times webpack also mentioned not capitalized.

### Configuring Jest to find our files

Now that Jest knows how to process our files, we need to tell it how to _find_ them. For Webpack's `modules` and `extensions` options there are direct analogs in Jest's `moduleDirectories` and `moduleFileExtensions` options.

> ```json title="package.json"

Reviewer: Is there a reason to configure the things in package.json and not giving the code examples with the separated jest.config.(ts/js) file?

> > Note: `<rootDir>` is a special token that gets replaced by Jest with the root of your project. Most of the time this will be the folder where your `package.json` is located unless you specify a custom `rootDir` option in your configuration.

Reviewer 2: Perhaps this update could be a good excuse to add few missing admonitions? (;

```suggestion
::: note

`<rootDir>` is a special token that gets replaced by Jest with the root of your project. Most of the time this will be the folder where your `package.json` is located unless you specify a custom [`rootDir`](Configuration.md#rootdir-string) option in your configuration.

:::
```

}
```

That's it! Webpack is a complex and flexible tool, so you may have to make some adjustments to handle your specific application's needs. Luckily for most projects, Jest should be more than flexible enough to handle your Webpack config.

> > Note: For more complex Webpack configurations, you may also want to investigate projects such as: [babel-plugin-webpack-loaders](https://github.com/istarkov/babel-plugin-webpack-loaders).

Reviewer 2: ```suggestion
:::tip

For more complex Webpack configurations, you may also want to investigate projects such as: [babel-plugin-webpack-loaders](https://github.com/istarkov/babel-plugin-webpack-loaders).

:::
```

## Using with Webpack

In addition to installing `babel-jest` as described earlier, you'll need to add `@babel/preset-env` like so:

> ```
> yarn add @babel/preset-env --dev

Reviewer 2: ```suggestion
```bash npm2yarn
npm install --save-dev @babel/preset-env
```
```

```

Then, you'll want to configure Babel as follows:

```json title=".babelrc"
{
  "presets": ["@babel/preset-env"]
}
```

> > Note: Jest caches files to speed up test execution. If you updated .babelrc and Jest is still not working, try running Jest with `--no-cache`.

Reviewer 2: ```suggestion
:::tip

Jest caches files to speed up test execution. If you updated `.babelrc` and Jest is still not working, try running `jest --clearCache`.

:::
```
